src/main/java/com/google/sps/data/Attendee.java

```diff
@@ -0,0 +1,91 @@
+package com.google.sps.data;
+
+import java.util.Date;
+import com.google.appengine.api.datastore.Entity;
+import com.google.sps.data.EntityConstants;
+
+/** Class that represents a user in the session as an attendee. */
+public class Attendee implements AttendeeInterface{
+
+  // The id of the session this attendee is in.
+  private final String sessionId;
+
+  // The screen name of this attendee.
+  private final String screenName;
+
+  // Time the attendee last polled the server.
+  private Date timeLastPolled;
+  
+  /** Initializes an Attendee object
+   * @param {String} sessionId - the session the attendee is attached to.
+   * @param {String} screenName - the screen name of the attendee. 
+   * @param {Date} timeLastPolled - the time attendee was last polled.
+   */
+  public Attendee (String sessionId, String screenName, Date timeLastPolled) {
+    this.sessionId = sessionId;
+    this.screenName = screenName;
+    this.timeLastPolled = timeLastPolled;
+  }
+
+  public String getSessionId() {
+    return sessionId;
+  }
+
+  public String getScreenName() {
+    return screenName;
+  }
+
+  public Date getTimeLastPolled() {
+    return timeLastPolled;
+  }
+
+  /** Compares two objects to see if the values are the same. */
+  public boolean isEqual(Object obj) {
```

Comment:
The proper way to do this (although I dont think we'll need it) this to override the equals() function of the Object class. That is the standard way to implement equality for any Java object. What you want is:

// Note that this does won't exist in your interface. This is an override of the Object equals()
public boolean equals(Object obj) {
   // self check
    if (this == obj)
      return true; 
    // null check 
    if (obj == null) 
      return false;
    // type check
    if(obj.getClass() != getClass()) 
      return false;
    return this.isEqual((Attendee) obj);
} 

// This exists in your interface
public boolean isEqual(AttendeeInterface attendee) {
...
}

Reply:
should the Override be in addition to the IsEqual function?

Reply:
yeah, technically all these functions should have override since this class is overriding some interface definition.